src/main/java/ch/njol/skript/expressions/ExprFormatDate.java

```diff
@@ -0,0 +1,137 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import java.text.SimpleDateFormat;
+
+import ch.njol.skript.lang.Literal;
+import ch.njol.skript.lang.VariableString;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.PropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.util.Date;
+import ch.njol.skript.util.Getter;
+import ch.njol.util.Kleenean;
+
+@Name("Formatted Date")
+@Description({
+	"Converts date to human-readable text format. By default, 'yyyy-MM-dd HH:mm:ss z' (e.g. '2018-03-30 16:03:12 +01') will be used. For reference, see this "
+		+ "<a href=\"https://en.wikipedia.org/wiki/ISO_8601\">Wikipedia article</a>."
+})
+@Examples({
+	"command /date:",
+	"\ttrigger:",
+	"\t\tsend \"Full date: %now formatted human-readable%\" to sender",
+	"\t\tsend \"Short date: %now formatted as \"\"yyyy-MM-dd\"\"%\" to sender"
+})
+@Since("2.2-dev31, INSERT VERSION (support variables in format)")
+public class ExprFormatDate extends PropertyExpression<Date, String> {
+	
+	private static final SimpleDateFormat DEFAULT_FORMAT = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss z");
+	
+	static {
+		Skript.registerExpression(ExprFormatDate.class, String.class, ExpressionType.PROPERTY,
+			"%dates% formatted [human-readable] [(with|as) %-string%]",
+			"[human-readable] formatted %dates% [(with|as) %-string%]");
+	}
+
+	@SuppressWarnings("NotNullFieldNotInitialized")
+	private SimpleDateFormat format;
+
+	@SuppressWarnings("NotNullFieldNotInitialized")
+	private Expression<? extends String> customFormat;
```

Comment:
```suggestion
	private Expression<String> customFormat;
```
String is final, no class can extend it